src/Aspire.Dashboard/wwwroot/js/app.js

```diff
@@ -314,26 +314,41 @@ function isInputElement(element, isRoot, isShadowRoot) {
 }
 
 window.registerGlobalKeydownListener = function(shortcutManager) {
-    const serializeEvent = function (e) {
-        if (e) {
-            return {
-                key: e.key,
-                code: e.keyCode.toString(),
-                location: e.location,
-                repeat: e.repeat,
-                ctrlKey: e.ctrlKey,
-                shiftKey: e.shiftKey,
-                altKey: e.altKey,
-                metaKey: e.metaKey,
-                type: e.type
-            };
-        }
-    };
+    function hasNoModifiers(keyboardEvent) {
+        return !keyboardEvent.altKey && !keyboardEvent.ctrlKey && !keyboardEvent.metaKey && !keyboardEvent.shiftKey;
+    }
+
+    function isOnlyShiftPressed(keyboardEvent) {
+        return keyboardEvent.shiftKey && !keyboardEvent.altKey && !keyboardEvent.ctrlKey && !keyboardEvent.metaKey;
+    }
 
     const keydownListener = function (e) {
-        if (!isActiveElementInput()) {
-            shortcutManager.invokeMethodAsync('OnGlobalKeyDown', serializeEvent(e));
+        if (isActiveElementInput()) return;
+        // list of shortcut enum codes is in src/Aspire.Dashboard/Model/IGlobalKeydownListener.cs
+        // to serialize an enum from js->dotnet, we must pass the enum's integer value, not its name
+        let shortcut = null;
+
+        if (isOnlyShiftPressed(e)) {
+            /* general shortcuts */
+            if (e.key === "?") shortcut = 100; // help
```

Comment:
use case statiment instead of many if/else statements?

Reply:
I don't prefer switch statements in JS, but do not have a strong opinion. Changed